astronomer/providers/apache/spark/hooks/spark_submit.py

```diff
@@ -0,0 +1,157 @@
+
+        spark_cmd = " ".join(spark_submit_cmd)
+
+        self._submit_sp = await asyncio.create_subprocess_shell(
+            spark_cmd,
+            stdout=asyncio.subprocess.PIPE,
+            stderr=asyncio.subprocess.PIPE,
+            bufsize=0,
+            **kwargs,
+        )
+        self.log.debug("Started subprocess: %r", self._submit_sp)
+        await asyncio.wait(
+            [
+                asyncio.create_task(self._process_submit_log(self._submit_sp.stdout)),
+                asyncio.create_task(self._process_submit_log(self._submit_sp.stderr)),
+            ]
+        )
+        return_code = self._submit_sp.returncode
+
+        if return_code:
+            raise AirflowException(
+                f"Cannot execute: {self._mask_cmd(spark_submit_cmd)}. Error code is: {return_code}."
+            )
+        if (
+            return_code is None
+            and self._connection["master"] == "local[*]"
+            and self._should_track_driver_status is False
+        ):
+            self._driver_status = "FINISHED"
+
+    async def _process_submit_log(self, stream: Any) -> None:
+        """
+        Processes the log and extracts useful information out of it.
+
+        :param stream: subprocess stream bytes
+        """
+        while True:
+            line = await stream.readline()
+            if line:
+                line = line.decode().strip()
+                line = line.strip()
+                self.log.info(line)
+                # TODO
+                # Need to implement the cluster mode yarn, standalone cluster mode, kubernetes
+                # Need to get the application id if cluster_mode is in yarn mode
+                # Need to get the driver id if cluster_mode is in standalone and kubernetes
+            else:
+                break
+
+    def track_driver_status(self) -> Optional[str]:
+        """Track driver status by driver id"""
+        if self._should_track_driver_status:
+            if self._driver_id is None:
+                raise AirflowException(
+                    "No driver id is known: something went wrong when executing the spark submit command"
+                )
+            # TODO
+            # Need to implement the cluster mode yarn, standalone cluster mode, kubernetes
+        return self._driver_status
+
+    def still_running(self) -> bool:
+        """
+        Check for the driver status if status not in ["FINISHED", "UNKNOWN", "KILLED", "FAILED", "ERROR"]
+         will return true, if not false which is considered the job is still running
+        """
+        driver_status = self.track_driver_status()
+        if driver_status not in ["FINISHED", "UNKNOWN", "KILLED", "FAILED", "ERROR"]:
+            return True
+        return False
+
+    def get_driver_status(self) -> Dict[str, Optional[str]]:
+        """Get the driver status and based on the driver status returns the message and status"""
+        msg = ""
+        if self._driver_status == "SUBMITTED":
```

Comment:
This is also ok, I'm just thinking if we can use some pattern matching like https://docs.python.org/3.10/whatsnew/3.10.html#pep-634-structural-pattern-matching

Reply:
python doesn't have any switch case by default it and elif acts as switch case in python, whatever you suggested I am not sure on the lower python version

Reply:
ok, then let's keep it as it is now.

Reply:
We can have a dictionary/map of status to message and then get the value for the _driver_status key from the map just in case if we want to avoid so many if-elifs. Also, make it easier to add/remove keys from the map and makes it quick & easier to read.